torchdata/nodes/epoch_updater.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
Can we keep this in the MultiNodeWeightedSampler instead of as a separate node?  I'm not sure we're ready to break it out yet